src/FSharpPlus/Control/Traversable.fs

```diff
@@ -49,15 +47,9 @@ type Traverse =
        Seq.foldBack cons_f t (result Seq.empty)
     #endif
 
-    static member Traverse (t: 't seq, f: 't->'u option, [<Optional>]_output: option<seq<'u>>, [<Optional>]_impl: Default2) =
-       let ok = ref true
-       let res = Seq.toArray (seq {
-           use e = t.GetEnumerator ()
-           while e.MoveNext() && ok.Value do
-               match f e.Current with
-               | Some v -> yield v
-               | None   -> ok.Value <- false})
-       if ok.Value then Some (Array.toSeq res) else None
+    static member inline Traverse (t: seq<'T>, f: 'T->'``Functor<'U>``, [<Optional>]_output: '``Functor<seq<'U>>``, [<Optional>]_impl: Default2) =
+        let mapped = Seq.map f t
+        Sequence.ForInfiniteSequences (mapped, IsLeftZeroForApply.Invoke) : '``Functor<seq<'U>>``
```

Comment:
The `Sequence.ForInfiniteSequences` looks a bit odd now that I'm looking at it. Though it seems to do the same thing as the left side. Why `ForInfiniteSequences` when it does to array?